lib/Models/WebMapServiceCatalogItem.js

```diff
@@ -1143,6 +1156,40 @@ WebMapServiceCatalogItem.prototype.exportData = function() {
     callWebCoverageService(this);
 };
 
+WebMapServiceCatalogItem.prototype.searchDataForMatchingFeatures = function(searchText, regex) {
+    const that = this;
+    return new Promise(function(resolve, reject) {
+        let workbenchItemResults = [];
+        if (that.hasLinkedWfs === false) {
+           resolve(workbenchItemResults);
+        } else if (that.hasLinkedWfs) {
+            that.linkedWfs.searchDataForMatchingFeatures(searchText, regex).then(function (res) {
+                workbenchItemResults = convertWfsSearchResultsToWms(res, that);
+                resolve(workbenchItemResults);
+            });
+        } else if (!defined(that.hasLinkedWfs)) {
+            loadXML(that.getCapabilitiesUrl.replace(/wms/gi, 'wfs'))
+            .then(function(xml) {
+                that.linkedWfs = new WebFeatureServiceCatalogItem(that.terria);
+                that.linkedWfs.url = cleanUrl(that.getCapabilitiesUrl.replace(/wms/gi, 'wfs'));
+                that.linkedWfs.typeNames = that.layers;
+                that.linkedWfs.requestGeoJson = true;
+                that.linkedWfs.requestGml = true;
+                that.linkedWfs._load().then(function (result) {
+                that.hasLinkedWfs = true;
+                that.linkedWfs.searchDataForMatchingFeatures(searchText, regex).then(function (res) {
+                    workbenchItemResults = convertWfsSearchResultsToWms(res, that);
+                    resolve(workbenchItemResults);
+                });
+                })
+                .catch(function (err) {
```

Comment:
This is probably crashing, it's called `.otherwise` in when.js promises, which `loadXML` returns.
In the current version of TerriaJS, we use Cesium's when.js promises everywhere. If we happen to get another kind of promise from somewhere, we immediately turn it into a when.js promise.

In the new model architecture (`mobx` branch), the rule is flipped. Native promises everywhere, convert anything else to a native promise immediately.